File: python/prime.py
class Sieve:
  def __init__(self, prime):
    self.prime = prime
    self.next = None
  
  def test(self, n):
    if self.prime * int(n / self.prime) == n:
      return None
    elif self.next == None:
      print(n)
      self.next = Sieve(n)
    else:
      self.next.test(n)

def main():
  s = Sieve(2);

  print("2");

  for i in range(2, 10):
      s.test(i)
  
if __name__== "__main__":
  main()

File: ruby/prime.rb
#!/usr/bin/env ruby

class Sieve
  @prime
  @next

  def initialize(prime)
    @prime = prime
  end

  def test(n)
    if @prime * (n / @prime) == n

    elsif @next.nil?
      puts(n)
      @next = Sieve.new(n)
    else
      @next.test(n)
    end
  end
end


def main
  s = Sieve.new(2)
  i = 2

  puts("2")

  while i < 13
    i += 1
    s.test(i)
  end
end

main